source/extensions/transport_sockets/tls/ocsp/ocsp.cc

```diff
@@ -265,6 +265,11 @@ ResponseData Asn1OcspUtility::parseResponseData(CBS& cbs) {
     throw EnvoyException("OCSP ResponseData is not a well-formed ASN.1 SEQUENCE");
   }
 
+  if (unwrap(Asn1Utility::getOptional(cbs, CBS_ASN1_CONTEXT_SPECIFIC | CBS_ASN1_CONSTRUCTED | 1)) ||
+      unwrap(Asn1Utility::getOptional(cbs, CBS_ASN1_CONTEXT_SPECIFIC | CBS_ASN1_CONSTRUCTED | 2))) {
+    throw EnvoyException("OCSP ResponseData version is not supported");
+  }
```

Comment:
By my reading of https://letsencrypt.org/docs/a-warm-welcome-to-asn1-and-der/#tag-classes, I think the trouble you're having here is that the tag for the Version field is `CBS_ASN1_CONTEXT_SPECIFIC | CBS_ASN1_CONSTRUCTED | 0` -- that is context-specific (0x80), constructed (0x20) tag 0 which is `0xA0`. The value returned must then be compared with 0, which indicates v1. If a different value is returned, Envoy should reject.

Reply:
It really helps me!